src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/TransactionExpirationQueueConsumerTests.kt

```diff
@@ -3258,4 +3282,333 @@ class TransactionExpirationQueueConsumerTests {
+        .willAnswer { npgResponse.fold({ Mono.just(it) }, { Mono.error(it) }) }
+
+      val errorContextCaptor = argumentCaptor<DeadLetterTracedQueueAsyncClient.ErrorContext>()
+      given(
+          deadLetterTracedQueueAsyncClient.sendAndTraceDeadLetterQueueEvent(
+            capture(binaryDataCaptor), errorContextCaptor.capture()))
+        .willReturn(mono {})
+
+      transactionExpirationQueueConsumer
+        .messageReceiver(
+          Either.right(
+            QueueEvent(
+              events.filterIsInstance<TransactionExpiredEvent>().first(), MOCK_TRACING_INFO)),
+          checkpointer,
+          MessageHeaders(mapOf()),
+        )
+        .block()
+
+      verify(transactionsRefundedEventStoreRepository, times(2)).save(any())
+      assertEquals(
+        1,
+        transactionRefundEventStoreCaptor.allValues
+          .filterIsInstance<TransactionRefundRequestedEvent>()
+          .count())
+      assertEquals(
+        1,
+        transactionRefundEventStoreCaptor.allValues
+          .filterIsInstance<TransactionRefundErrorEvent>()
+          .count())
+      verify(deadLetterTracedQueueAsyncClient, times(1))
+        .sendAndTraceDeadLetterQueueEvent(any(), any())
+      assertEquals(
+        DeadLetterTracedQueueAsyncClient.ErrorCategory.REFUND_MANUAL_CHECK_REQUIRED,
+        errorContextCaptor.firstValue.errorCategory)
+    }
+
+    @Test
+    fun `messageReceiver should emit refund retry event when get NPG transaction status returns transient error`() {
+      val events =
+        listOf(
+          transactionActivateEvent(
+            NpgTransactionGatewayActivationData(
+              UUID.randomUUID().toString(), UUID.randomUUID().toString())),
+          transactionAuthorizationRequestedEvent(
+            TransactionAuthorizationRequestData.PaymentGateway.NPG,
+            NpgTransactionGatewayAuthorizationRequestedData(
+              URI.create("http://localhost"),
+              "MC",
+              UUID.randomUUID().toString(),
+              UUID.randomUUID().toString())),
+          transactionExpiredEvent(TransactionStatusDto.EXPIRED))
+      setupTransactionStorageMock(events)
+      given(checkpointer.success()).willReturn(Mono.empty())
+      given { authorizationStateRetrieverService.getStateNpg(any(), any(), any(), any(), any()) }
+        .willAnswer { Mono.error<StateResponseDto>(NpgServerErrorException("Any 5xx error")) }
+
+      given { refundRetryService.enqueueRetryEvent(any(), any(), any()) }.willAnswer { mono {} }
+
+      transactionExpirationQueueConsumer
+        .messageReceiver(
+          Either.right(
+            QueueEvent(
+              events.filterIsInstance<TransactionExpiredEvent>().first(), MOCK_TRACING_INFO)),
+          checkpointer,
+          MessageHeaders(mapOf()),
+        )
+        .block()
+
+      verify(transactionsRefundedEventStoreRepository, times(2)).save(any())
```

Comment:
add assertion to verify that transaction-view is updated with the expected statuses (REFUND_REQUESTED and REFUND_ERROR)

Reply:
Solved by https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/255/commits/a03dde092c41a59ff7dcefdfb6f0a5b0a4585274